orchagent/bufferorch.cpp

```diff
@@ -82,6 +97,38 @@ void BufferOrch::initBufferReadyList(Table& table)
     }
 }
 
+void BufferOrch::initFlexCounterGroupTable(void)
+{
+    string bufferPoolWmPluginName = "watermark_bufferpool.lua";
+
+    try
+    {
+        string bufferPoolLuaScript = swss::loadLuaScript(bufferPoolWmPluginName);
+        string bufferPoolWmSha = swss::loadRedisScript(m_countersDb.get(), bufferPoolLuaScript);
+
+        vector<FieldValueTuple> fvTuples;
+        fvTuples.emplace_back(BUFFER_POOL_PLUGIN_FIELD, bufferPoolWmSha);
+        fvTuples.emplace_back(POLL_INTERVAL_FIELD, BUFFER_POOL_WATERMARK_FLEX_STAT_COUNTER_POLL_MSECS);
+
+        string statsMode = STATS_MODE_READ_AND_CLEAR;
+        // Some platforms do not support buffer pool watermark clear operation
+        // Need the SAI API support to query the capability
+        // Before this capability is in place, we feed the platform info into the orchagent
+        // daemon and check against it
+        char *device = getenv("device");
+        if (device && (strstr(device, "7050") || strstr(device, "6000")))
+        {
+            statsMode = STATS_MODE_READ;
+        }
+        fvTuples.emplace_back(STATS_MODE_FIELD, statsMode);
+        m_flexCounterGroupTable->set(BUFFER_POOL_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP, fvTuples);
+    }
+    catch (...)
```

Comment:
>catch (...) [](start = 4, length = 11)

Dangerous to catch all.

Reply:
Still too general, could you be more specific?

---
In reply to: [277914027](https://github.com/Azure/sonic-swss/pull/853#discussion_r277914027) [](ancestors = 277914027)